tools/hxcpp/Setup.hx

```diff
@@ -5,82 +5,83 @@ import BuildTool;
 
 class Setup
 {
-   static function findAndroidNdkRoot(inDir:String, inBaseVersion:Int)
+   static function findAndroidNdkRoot(defines: Map<String,String>, inBaseVersion:Int):String
    {
-      var files:Array<String> = null;
-      try
-      {
-         files = FileSystem.readDirectory(inDir);
-      }
-      catch (e:Dynamic)
-      {
-         Log.error('ANDROID_NDK_DIR "$inDir" does not point to a valid directory');
-         //throw 'ANDROID_NDK_DIR "$inDir" does not point to a valid directory.';
-      }
-
-      var extract_version = ~/^(android-ndk-)?r(\d+)([a-z]?)$/;
-      var bestMajor = 0;
-      var bestMinor = "";
-      var result = "";
-      for(file in files)
-         if (extract_version.match(file))
-         {
-            var major = Std.parseInt( extract_version.matched(2) );
-            if (inBaseVersion==0 || major==inBaseVersion)
-            {
-               var minor = extract_version.matched(3);
-               if ( major>bestMajor || (major==bestMajor && minor>bestMinor))
-               {
-                  bestMajor = major;
-                  bestMinor = minor;
-                  result = inDir + "/" + file;
-               }
-            }
-         }
-
-      Log.info("", "Found NDK " + result);
-
-      if (result=="")
-      {
-         if (inBaseVersion!=0)
-            Log.error('ANDROID_NDK_DIR "$inDir" does not contain requested NDK $inBaseVersion');
-         else
-            Log.error('ANDROID_NDK_DIR "$inDir" does not contain a matching NDK');
-         //throw 'ANDROID_NDK_DIR "$inDir" does not contain matching ndk downloads.';
+      var bestVersion = 0.0;
+      var result:String = null;
+
+      Log.v("Looking in ANDROID_NDK_DIR");
+      if (defines.exists("ANDROID_NDK_DIR"))
+      {
+        var ndkDir:String = defines.get("ANDROID_NDK_DIR");
+        ndkDir = ndkDir.split("\\").join("/");
+        var files:Array<String> = null;
+        var checkFiles:Bool = true;
+        try
+        {
+             files = FileSystem.readDirectory(ndkDir);
+        }
+        catch (e:Dynamic)
+        {
+             Log.warn('ANDROID_NDK_DIR "$ndkDir" does not point to a valid directory');
+             checkFiles=false;
+        }
+        if (checkFiles)

```

Comment:
Is this bool needed anymore? Seems like you could just call a function to check the files inside of the catch.

Reply:
Initialy this code only considered the name of directories to detect the NDK version, and later added to check source.properties file.  The check is done in 2 places, but now it calls getNdkVersion on both. Things start to be a little tricky when trying to use the NDKVXX flag. I believe this code fixes it when retrieving the version from the source.properties file and also gives a higher prority than ANDROID_NDK_ROOT. It also checks in the default Android studio ndk-bundle directory.

Reply:
Yes, maybe it would be the same as writting the checkFiles block inside the "try"